RadioControl/RadioControl.ino

```diff
@@ -0,0 +1,329 @@
+void sendElcanoSerial(void);
+void debugPacket(unsigned long timer);
+
+#define ESERIAL Serial1
+#define PC Serial
+
+#define ProcessFallOfINT(Index) RC_elapsed[Index] = (micros() - RC_rise[Index])
+#define ProcessRiseOfINT(Index) RC_rise[Index] = micros()
+#define RC_NUM_SIGNALS 7 // How many RC signals we receive
+volatile unsigned long RC_rise[RC_NUM_SIGNALS];
+volatile unsigned long RC_elapsed[RC_NUM_SIGNALS];
+/* RC_Done[] tells us when we have started receiving RC data. Until then, we
+  ignore RC_rise and RC_elapsed. */
+volatile bool RC_Done[RC_NUM_SIGNALS];
+volatile bool synced = false;
+volatile bool flipping;
+#define INVALID_DATA 0L // This is a value that the RC controller can't produce.
+volatile int rc_index = 0;
+
+const int MSG_SIZE = 3;
+static bool using_us_units = false; // we use byte values(0-255) or microseconds (~1000-2000us)
+
+RH_ASK driver;
+/* default config:
+  RH_ASK(uint16_t speed = 2000, uint8_t rxPin = 11, uint8_t txPin = 12, uint8_t pttPin = 10, bool pttInverted = false); */
+
+/* LEDs
+  analogWrite works on pins 3, 5*, 6*, 9, 10, and 11 of a Micro. * = higher duty cycle */
+const uint8_t TX_LED_TURN = 3;
+const uint8_t TX_LED_THROTTLE = 5;
+const uint8_t TX_LED_POWER = 6;
+const uint8_t TX_LED_AUTO = 7;
+const uint8_t TX_LED_EBRAKE = 8;
+const uint8_t RESERVED_PIN = 10;    // this is the enable transmitter pin in the default radiohaed config
+const uint8_t TX_LED_TRANSMIT = 11; // D11 is also radiohead RX pin but we are not using it on the transmitter
+const uint8_t RX_LED_NOSIGNAL = 12; // D60 is A6 on the Mega
+
+void setupReceiver() {
+  driver.init();
+  driver.setModeRx();
+  PC.begin(baudrate);
+  ESERIAL.begin(baudrate);
+}
+
+void waitInputByte(int timeout) {
+  if (DEBUG) {
+    PC.println(F("waiting for radio packet..."));
+  }
+  inputData.newData = false;
+  static uint8_t buf[MSG_SIZE] = {0};
+  uint8_t buflen = sizeof(buf);
+  do {
+    if (driver.recv(buf, &buflen)) {
+      if (DEBUG) {
+        PC.print(millis());
+      }
+      if (DEBUG) {
+        PC.println(F(" packet found"));
+      }
+      // we do have a valid packet so we pull out the data and set newData
+      inputData.throttle = buf[0];
+      inputData.angle = buf[1];
+      inputData.autoOn = bitRead(buf[2], 1);
+      inputData.ebrakeOn = bitRead(buf[2], 2);
+      inputData.newData = true;
+    }
+    if (timeout > 0) {
+      delay(1);
+      timeout--;
+    }
+  } while (timeout > 0);
+}
+
+void waitInputPWM(int timeout) {
+  inputData.newData = false;
+  for (int i = 0; i < RC_NUM_SIGNALS; i++) {
+    RC_rise[i] = INVALID_DATA;
+    RC_elapsed[i] = INVALID_DATA;
+    RC_Done[i] = 0;
+  }
+  attachInterrupt(digitalPinToInterrupt(IRPT_TURN), ISR_TURN_rise, RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_GO), ISR_GO_rise, RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_ESTOP), ISR_ESTOP_rise, RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_BRAKE), ISR_BRAKE_rise, RISING);
```

Comment:
We talked about these -- they are being moved over from C2, but this is in progress.

Reply:
We're going to merge this PR, and then we can all have copies of it to play with.